lib/syskit/cli/log_runtime_archive.rb

```diff
@@ -41,6 +42,42 @@ def process_root_folder
                 end
             end
 
+            # Manages folder available space
+            #
+            # The method will check if there is enough space to save more log files
+            # according to pre-established threshold.
+            #
+            # @param [integer] free_space_low_limit: required free space threshold, at
+            #   which the archiver starts deleting the oldest log files
+            # @param [integer] free_space_delete_until: post-deletion free space, at which
+            #   the archiver stops deleting the oldest log files
+            def ensure_free_space(free_space_low_limit, free_space_delete_until)
+                if free_space_low_limit > free_space_delete_until
+                    raise ArgumentError,
+                          "cannot erase files: freed limit is smaller than " \
+                          "low limit space."
+                end
+
+                stat = Sys::Filesystem.stat(@target_dir)
+                available_space = stat.bytes_free
+
+                return if available_space > free_space_low_limit
+
+                until available_space >= free_space_delete_until
+                    files = @target_dir.each_child.select(&:file?)
+                    if files.empty?
+                        Roby.warn "Cannot erase files: the folder is empty but the "\
+                        "available space is smaller than the threshold."
+                        break
+                    end
+
+                    removed_file = files.min
```

Comment:
Are we going with removing the smaller file? I thought we were aiming to remove the oldest one, which was also the first one alphabetically given the way we name the logs.

Reply:
Yeah, it was a misconception from me. String do input the #<=> operator for comparing strings alphabetically.